### PR TITLE
Parsing ORCA MP2 gradients

### PR DESCRIPTION
**Pull request by alexmmaldonado**
The ORCA MP2 module uses a different flag other than "CARTESIAN GRADIENT" for gradients. Here, I just added another block to the ORCA parser that catches these MP2 gradients.

### CODE REVIEW

File: cclib/parser/orcaparser.py [hunk 1]
```diff
@@ -403,6 +403,25 @@ def splitter(line):
             if not hasattr(self, 'grads'):
                 self.grads = []
             self.grads.append(grads)
+        
+        # Grab MP2 gradients
```
Comment (shivupa) [inline cclib/parser/orcaparser.py]: This involves a little bit of code duplication. We could fix that by doing the following.

The `next` statements at 392 and 393 aren't rewriting `line` so at 394 we could add `if line[:18] == 'CARTESIAN GRADIENT' or line[:22] == 'The final MP2 gradient':` 

Then lines 413-424 could be removed.

Remember to take the comment here on 407 the comment and combine it with the comment on 390

Let me know if this makes sense

**Comment by shivupa** [comment]
Thanks for this! I had a minor change. Also we don't currently have a test that tests this. Do you have an example that we could turn into a test? The way this would work would be to add the file to the relevant directory [here](https://github.com/cclib/cclib-data) and add a test in [here](https://github.com/cclib/cclib/blob/master/test/regression.py) that tests that the gradient is parsed correctly. Let me know if this makes sense.

File: cclib/parser/orcaparser.py (hunk 1 above)
Comment (alexmmaldonado) [reply cclib/parser/orcaparser.py]: Yes, that makes sense for the most part.

I combined the two signals for cartesian gradient and MP2 gradient and merged the common code. Let me know if this is what you were looking for, or what else you would like me to correct.

File: cclib/parser/orcaparser.py [hunk 2]
```diff
@@ -388,22 +388,39 @@ def splitter(line):
...
 
             grads = []
-            line = next(inputfile).strip()
-            while line:
-                idx, atom, colon, x, y, z = line.split()
-                grads.append((float(x), float(y), float(z)))
 
+            if line[:18] == 'CARTESIAN GRADIENT':
+                next(inputfile)
+                next(inputfile)
```
Comment (berquist) [inline cclib/parser/orcaparser.py]: I understand that the original code had these lines, but can you convert them to use `skip_lines`?

File: cclib/parser/orcaparser.py [hunk 3]
```diff
@@ -388,22 +388,39 @@ def splitter(line):
...
+
+                line = next(inputfile).strip()
+                while line:
+                    idx, atom, colon, x, y, z = line.split()
```
Comment (berquist) [inline cclib/parser/orcaparser.py]: You could do something like
```python
tokens = line.split()
x, y, z = float(tokens[-3]), float(tokens[-2]), float(tokens[-1])
```
and then both `while` blocks contain the same code, and you can combine the two `if line[...` blocks.

File: cclib/parser/orcaparser.py (hunk 2 above)
Comment (alexmmaldonado) [reply cclib/parser/orcaparser.py]: Not a problem.

File: cclib/parser/orcaparser.py (hunk 3 above)
Comment (alexmmaldonado) [reply cclib/parser/orcaparser.py]: Yes, that works great.

**Comment by alexmmaldonado** [comment]
I added, to the best of my knowledge, the regression test to the output file I added to cclib-data. Let me know if there is something I am missing.

**Pull request merged**